cocotb/clock.py

```diff
@@ -65,12 +65,43 @@ class Clock(BaseClock):
         signal: The clock pin/signal to be driven.
         period (int): The clock period. Must convert to an even number of
             timesteps.
-        units (str, optional): One of 
+        units (str, optional): One of
             ``None``, ``'fs'``, ``'ps'``, ``'ns'``, ``'us'``, ``'ms'``, ``'sec'``.
             When no *units* is given (``None``) the timestep is determined by
             the simulator.
+
+
+    If you need more features like a phase shift and an asymmetric duty cycle,
+    it is simple to create your own clock generator (that you then :func:`fork`):
+
+    .. code-block:: python
+
+        @cocotb.coroutine
+        def custom_clock():
+            high_time = Timer(high_delay)
+            low_time = Timer(low_delay)
+            yield Timer(initial_delay)
```

Comment:
Perhaps add `# pre-construct triggers for performance`

Reply:
Done.